homeassistant/components/bkk/sensor.py

```diff
@@ -0,0 +1,311 @@
+"""BKK Futar integration to monitor public transport stops of Budapest city."""
+
+from datetime import datetime, timedelta
+import logging
+
+import futar
+import voluptuous as vol
+
+from homeassistant.components import group
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import EVENT_HOMEASSISTANT_START, STATE_UNAVAILABLE
+from homeassistant.helpers import entity_registry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = "bkk"
+CONF_STOP = "stop"
+CONF_GROUPED = "grouped"
+CONF_ONLY_DEPARTURES = "only_departures"
+CONF_MINUTES_AFTER = "minutes_after"
+MAX_DEPARTURES = 5
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_STOP): cv.string,
+        vol.Optional(CONF_GROUPED, default=False): cv.boolean,
+        vol.Optional(CONF_ONLY_DEPARTURES, default=True): cv.boolean,
+        vol.Optional(CONF_MINUTES_AFTER, default=120): cv.positive_int,
+    }
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ICONS = {
+    "BUS": "mdi:bus",
+    "TRAM": "mdi:tram-side",
+    "RAIL": "mdi:tram",
+    "SUBWAY": "mdi:subway",
+    "API": "mdi:api",
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_create_group(hass, name, entities):
+    """Create group for bkk route entities."""
+    await group.Group.async_create_group(hass, name, entities, user_defined=True)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up bkk platform."""
+    data_updater = BkkDataUpdater(config[CONF_STOP],
+                                  config[CONF_GROUPED],
+                                  config[CONF_ONLY_DEPARTURES],
+                                  config[CONF_MINUTES_AFTER])
+    try:
+        data_updater.get_data()
+    except ConnectionError:
+        return
+    else:
+        if data_updater.state == "NOT_FOUND":
+            _LOGGER.error(
+                "Stop id could not be found. Please specify an existing BKK stop id"
+            )
+            return
+        entities = []
+        stop_type = data_updater.data["references"]["stops"][data_updater.stop_id][
+            "type"
+        ]
+        for i in range(MAX_DEPARTURES):
+            route = BkkRoute(data_updater.stop_id, i, stop_type)
+            entities.append(route)
+        data_updater.routes = entities
+        async_add_entities(entities, True)
+        async_add_entities([data_updater], True)
+
+        async def async_remove_unavailable_entities(event):
```

Comment:
Why do we need to remove unavailable entities?

Reply:
I just found that home assistant did not start with a clean state. There were unused unavailable entities from previous runs, so I decided to make a cleanup first. I intended to remove only those entities that belong to my domain.

Reply:
Home Assistant will be default restore entities from the entity registry as unavailable. If they still are unavailable after home assistant start up it means that the integration doesn't update their state.

If the integration sets a `unique_id` for the entites, ie activates the entity registry, make sure to update the entities during the set up phase of home assistant. This will make the entities show with the correct state, ie not unavailable.